src/Npgsql/NpgsqlConnection.cs

```diff
@@ -2044,6 +2049,9 @@ public void ReloadTypes()
     {
         CheckReady();
         using var scope = StartTemporaryBindingScope(out var connector);
+        // LoadDatabaseInfo might attempt to execute a query over a connector, which might run in parallel to KeepAlive.
```

Comment:
Not sure the comment is even necessary - there could also be accidental concurrent usage with another user execution or whatever (as always).